vulture/core.py

```diff
@@ -68,6 +69,33 @@ class VultureInputException(Exception):
     pass
 
 
+# Reinventing wheel here to keep vulture lightweight
+def memoize(func):
+    _cache = {}
+
+    @wraps(func)
+    def wrapped(*args):
+        try:
+            return _cache[args]
+        except KeyError:
+            _cache[args] = result = func(*args)
+            return result
+    return wrapped
+
+
+@memoize
+def weigh(node):
+    """
+    Recursively count child AST nodes under `node`. It is an approximation
+    of the amount of code belonging to the node, which is useful for
+    sorting the list of unused code that a developer might want to remove
+    """
+    if isinstance(node, (ast.FunctionDef, ast.If, ast.For, ast.ClassDef)):
```

Comment:
I analyzed which AST node types have which fields with a little script:

```
import ast

for type_name in sorted(dir(ast)):
    print type_name,
    typ = getattr(ast, type_name)
    try:
        print typ._fields
    except AttributeError:
        pass
```
Filtering the types that span multiple lines, we get:

```
ClassDef ('name', 'bases', 'body', 'decorator_list')
ExceptHandler ('type', 'name', 'body')
For ('target', 'iter', 'body', 'orelse')
FunctionDef ('name', 'args', 'body', 'decorator_list')
If ('test', 'body', 'orelse')
Module ('body',)
TryExcept ('body', 'handlers', 'orelse')
TryFinally ('body', 'finalbody')
While ('test', 'body', 'orelse')
With ('context_expr', 'optional_vars', 'body')
```
Removing the fields that don't add additional lines, we get:

```
ClassDef ('body', 'decorator_list')
ExceptHandler ('body')
For ('body', 'orelse')
FunctionDef ('body', 'decorator_list')
If ('body', 'orelse')
Module ('body',)
TryExcept ('body', 'handlers', 'orelse')
TryFinally ('body', 'finalbody')
While ('body', 'orelse')
With ('body')
```

I think traversing over these nodes should make the line estimates more accurate. Could you extend the code in that direction?